python_modules/dagster/dagster/_generate/templates/PROJECT_NAME_PLACEHOLDER/requirements.txt.tmpl

```diff
@@ -0,0 +1,3 @@
+-e {{ repo_name }}
```

Comment:
I haven't thought about this deeply, but if we're including setup.py at the repository level, I would expect requirements.txt to be at that level as well?

Reply:
for the minimal case, they are at the same level, and it doesn't matter.

the reason why requirements.txt is in project root while setup.py is at the repo level is for the multi repo case (example: https://github.com/yuhan/my-dagster-project-multiple-locations), we'll have the requirements.txt at the root which installs all the dagster repos / code locations as separate python packages: 
```
.
├── README.md
├── cloud_workspace.yaml
├── my_repo_1
│   ├── my_repo_1
│   │   ├── __init__.py
│   │   ├── assets
│   │   │   └── __init__.py
│   │   └── repository.py
│   ...
│   ├── pyproject.toml
│   ├── setup.cfg
│   └── setup.py
├── my_repo_2
│   ├── my_repo_2
│   │   ├── __init__.py
│   │   ├── assets
│   │   │   └── __init__.py
│   │   └── repository.py
│   ...
│   ├── pyproject.toml
│   ├── setup.cfg
│   └── setup.py
├── requirements.txt
```

Currently, we want the requirements.txt in the git repo root because that's how the cloud cli assumes. 
I don't think we have a final decision yet. We're still discussing that with cloud:
https://elementl-workspace.slack.com/archives/C03MCLC0L83/p1659142575405289?thread_ts=1659137503.267199&cid=C03MCLC0L83

Reply:
in the multi-repo case, each entry in the requirements.txt would be one of the code locations?

Reply:
yes, it'd be
```
my_repo_1
my_repo_2
# and other reqs for deployment
```

Reply:
In many cases, they'll have conflicting dependencies, so they can't be installed together into the same python environment. But the point of a requirements.txt is a set of libraries that go into the same python environment, right?

Reply:
ah that's right - confirmed w/ cloud that each code location and have its own requirements.txt for serverless. 
moved requirements.txt to repo level